src/eradiate/scenes/atmosphere/_heterogeneous.py

```diff
@@ -363,10 +364,21 @@ def kernel_phase(self, ctx: KernelDictContext) -> KernelDict:
                 sigma_ss.append(radprops.sigma_t * radprops.albedo)
 
             # Construct a blended phase function based on those weighting values
+            shape = self.eval_shape(ctx)
+            if isinstance(shape, CuboidShape):
+                shape_min = shape.center - shape.edges * 0.5
+                shape_min[2] = self.bottom
+                shape_max = shape.center + shape.edges * 0.5
+            elif isinstance(shape, SphereShape):
+                length_units = ucc.get("length")
+                shape_min = [0, 0, 0] * length_units
+                shape_max = [1, 1, shape.radius.m_as(length_units)] * length_units
```

Comment:
I must admit: I haven't tested this with the spherical shell geometry. Are you certain that this will be correct? If not, I'd rather have this raise.

Other shapes should be explicitly unsupported. Our current way to do so is to raise a `RuntimeError`:
```suggestion

            elif isinstance(shape, SphereShape):
                length_units = ucc.get("length")
                shape_min = [0, 0, 0] * length_units
                shape_max = [1, 1, shape.radius.m_as(length_units)] * length_units
                
            else:
                raise RuntimeError  # Only cuboids and spheres are allowed currently
```

Reply:
Other shapes are excluded by `eval_shape()`, I had a check here at first too, but it can actually not be anything else, I think.

Reply:
Having this `else` here is still free and indicates that special care must be taken when adding a new shape, it's important that this unhandled case raises (if `eval_shape()` is fixed but not this method, then something can go silently bad).